### PR TITLE
swift #7583: Support if let shorthand for shadowing an existing optional variable

### PR DESCRIPTION
**Pull request by ihji**
PR checklist:

- [x] Purpose of the code is [evident to future readers](https://semgrep.dev/docs/contributing/contributing-code/#explaining-code)
- [x] Tests included or PR comment includes a reproducible test plan
- [x] Documentation is up-to-date
- [x] A changelog entry was [added to changelog.d](https://semgrep.dev/docs/contributing/contributing-code/#adding-a-changelog-entry) for any user-facing change
- [x] Change has no security implications (otherwise, ping security team)

If you're unsure about any of this, please see:

- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

### CODE REVIEW

File: src/core/Lang.ml [hunk 1]
```diff
@@ -146,6 +146,7 @@ let langs_of_filename filename =
   | FT.PL FT.Rust -> [ Rust ]
   | FT.PL FT.R -> [ R ]
   | FT.PL FT.Scala -> [ Scala ]
+  | FT.PL FT.Swift -> [ Swift ]
```
Comment (nmote) [inline src/core/Lang.ml:149]: What happened here?

Comment (ihji) [reply src/core/Lang.ml:149]: Can't print out the ast by `-dump_ast` without it.

Comment (nmote) [reply src/core/Lang.ml:149]: Weird, I know I used `-dump_ast` for Swift all the time. Maybe something got messed up in a refactoring.

Comment (ihji) [reply src/core/Lang.ml:149]: Probably `-dump_tree_sitter_cst` 😓 Will check.

Comment (ihji) [reply src/core/Lang.ml:149]: It was `-dump_il` 😆

Comment (nmote) [reply src/core/Lang.ml:149]: Got it yeah, I don't think I ever used `-dump_il` for Swift.

**Review by nmote — APPROVED** [review]

**Pull request merged**